tt_metal/impl/dispatch/command_queue.hpp

```diff
@@ -522,6 +523,8 @@ class HWCommandQueue {
     // post trace capture.
     uint32_t multicast_cores_launch_message_wptr_reset = 0;
     uint32_t unicast_cores_launch_message_wptr_reset = 0;
+    // True if the next program to be enqueued must be stalled before writing memory.
+    bool must_early_stall_program = false;
```

Comment:
Please consider renaming this so that we know why the stall is being issued. Maybe something like: `issue_early_stall_after_trace`.

Reply:
PTAL. I've rewritten the change to handle this tracking in WorkerConfigBufferMgr.